contracts/staking/contracts/src/immutable/MixinHyperParameters.sol

```diff
@@ -0,0 +1,163 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
+import "@0x/contracts-utils/contracts/src/Ownable.sol";
+import "../interfaces/IStakingEvents.sol";
+import "../libs/LibStakingRichErrors.sol";
+import "./MixinConstants.sol";
+
+
+contract MixinHyperParameters is
+    IStakingEvents,
+    MixinConstants,
+    Ownable
+{
+    // Minimum seconds between epochs.
+    uint256 internal epochDurationInSeconds = 2 weeks;
+    // How much delegated stake is weighted vs operator stake, in ppm.
+    uint32 internal rewardDelegatedStakeWeight = (90 * PPM_DENOMINATOR) / 100; // 90%
+    // Minimum amount of stake required in a pool to collect rewards.
+    uint256 internal minimumPoolStake = 100 * MIN_TOKEN_VALUE; // 100 ZRX
+    // Maximum number of maker addresses allowed to be registered to a pool.
+    uint256 internal maximumMakersInPool = 10;
+    // Numerator for cobb douglas alpha factor.
+    uint256 internal cobbDouglasAlphaNumerator = 1;
+    // Denominator for cobb douglas alpha factor.
+    uint256 internal cobbDouglasAlphaDenomintor = 2;
+
+    /// @dev Set all hyperparameters at once.
+    /// @param _epochDurationInSeconds Minimum seconds between epochs.
+    /// @param _rewardDelegatedStakeWeight How much delegated stake is weighted vs operator stake, in ppm.
+    /// @param _minimumPoolStake Minimum amount of stake required in a pool to collect rewards.
+    /// @param _maximumMakersInPool Maximum number of maker addresses allowed to be registered to a pool.
+    /// @param _cobbDouglasAlphaNumerator Numerator for cobb douglas alpha factor.
+    /// @param _cobbDouglasAlphaDenomintor Denominator for cobb douglas alpha factor.
+    function tune(
```

Comment:
Nit - I think either `tuneHyperParameters` or `setHyperParameters` would be good to use here, as it'll provide context to the overall Staking API.